components/admin/requests/processing/TaskCardUploadStep.tsx

```diff
@@ -0,0 +1,122 @@
+import { SyntheticEvent, useRef, useState, FC } from 'react';
+import { WarningTwoIcon, CloseIcon } from '@chakra-ui/icons';
+import { Button, Text, Link, HStack, VStack, IconButton } from '@chakra-ui/react';
+
+// 5 MB file size limit
+const FILE_SIZE_LIMIT = 5 * 1024 * 1024;
+
+// File upload button props
+type Props = {
+  readonly file: File | null; // currently uploaded file
+  readonly onUploadFile: (selectedFile: File) => void; // handle file upload
+  readonly isDisabled: boolean;
+  readonly onUndo: () => void;
+  readonly documentUrl: string | null;
+};
+
+/**
+ * POA form upload component allowing users to upload POA form PDF file
+ */
+export const TaskCardUploadStep: FC<Props> = ({
+  isDisabled,
+  file,
+  onUploadFile,
+  onUndo,
+  documentUrl,
+}) => {
+  const hiddenFileInput = useRef<HTMLInputElement>(null);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+  /**
+   * Handle upload button click
+   */
+  const handleClick = () => {
+    if (hiddenFileInput.current) {
+      hiddenFileInput.current.click();
+    }
+  };
+
+  /**
+   * Handle file selection
+   */
+  const handleChange = (event: SyntheticEvent) => {
+    const target = event.target as HTMLInputElement;
+    const uploadedFile = (target.files as FileList)[0] || null;
+
+    if (!uploadedFile) {
+      return;
+    }
+
+    setErrorMessage(null);
+
+    // Verify that uploaded file is < 5MB
+    if (uploadedFile && uploadedFile.size > FILE_SIZE_LIMIT) {
+      setErrorMessage('File exceeds maximum size');
```

Comment:
should this also return so that the onUploadFile function is not called for invalid files?